media-libs/openexr/files/openexr-2.2.0-post-release-fixes-v20170109.patch

```diff
@@ -0,0 +1,386 @@
++                    // 9   11  18  24  31  40  44  53
++                    // 10  19  23  32  39  45  52  54
++                    // 20  22  33  38  46  51  55  60
++                    // 21  34  37  47  50  56  59  61
++                    // 35  36  48  49  57  58  62  63
+                     //
+                     // If lastNonZero is less than the first item on
+                     // each row, we know that the whole row is zero and 
+@@ -888,25 +888,25 @@ DwaCompressor::LossyDctDecoderBase::execute ()
+                     //
+                     // where:
+                     //
+-                    //    const int rowStartIdx[] = {2, 5, 9, 14, 20, 27, 35};
++                    //    const int rowStartIdx[] = {2, 3, 9, 10, 20, 21, 35};
+                     //    const int rowsEmpty[]   = {7, 6, 5,  4,  3,  2,  1};
+                     //
+ 
+                     if (lastNonZero < 2)
+                         dctInverse8x8_7(_dctData[comp]._buffer);
+-                    else if (lastNonZero < 5)
++                    else if (lastNonZero < 3)
+                         dctInverse8x8_6(_dctData[comp]._buffer);
+                     else if (lastNonZero < 9)
+                         dctInverse8x8_5(_dctData[comp]._buffer);
+-                    else if (lastNonZero < 14)
++                    else if (lastNonZero < 10)
+                         dctInverse8x8_4(_dctData[comp]._buffer);
+                     else if (lastNonZero < 20)
+                         dctInverse8x8_3(_dctData[comp]._buffer);
+-                    else if (lastNonZero < 27)
++                    else if (lastNonZero < 21)
+                         dctInverse8x8_2(_dctData[comp]._buffer);
+                     else if (lastNonZero < 35)
+                         dctInverse8x8_1(_dctData[comp]._buffer);
+-                    else
++                    else 
+                         dctInverse8x8_0(_dctData[comp]._buffer);
+                 }
+             }
+@@ -947,7 +947,7 @@ DwaCompressor::LossyDctDecoderBase::execute ()
+                 }
+                 else
+                 {
+-                    #if IMF_HAVE_SSE2
++                    #ifdef IMF_HAVE_SSE2
+ 
+                         __m128i *dst = (__m128i*)&rowBlock[comp][blockx*64];
+ 
+@@ -1922,7 +1922,7 @@ DwaCompressor::compress
+     if (outBufferSize > _outBufferSize) 
+     {
+         _outBufferSize = outBufferSize;
+-        if (_outBuffer == 0)
++        if (_outBuffer != 0)
+             delete[] _outBuffer;       
+         _outBuffer = new char[outBufferSize];
+     }
+diff --git a/IlmImf/ImfDwaCompressorSimd.h b/IlmImf/ImfDwaCompressorSimd.h
+index 93246f6..b07fa4e 100644
+--- a/IlmImf/ImfDwaCompressorSimd.h
++++ b/IlmImf/ImfDwaCompressorSimd.h
+@@ -334,34 +334,37 @@ interleaveByte2 (char *dst, char *src0, char *src1, int numBytes)
+         // use aligned loads
+         //
+     
+-        for (int x = 0; x < 8; ++x)
++        for (int x = 0; x < std::min (numBytes, 8); ++x)
```

Comment:
that kind of change seems independent of the cmakelists patches; please split it in logically independent patches